packages/transaction-controller/src/utils/gas.ts

```diff
@@ -200,6 +202,70 @@ export function addGasBuffer(
   return maxHex;
 }
 
+/**
+ * Simulate the required gas for a batch of transactions using the simulation API.
+ *
+ * @param options - The options object.
+ * @param options.chainId - The chain ID of the transactions.
+ * @param options.from - The address of the sender.
+ * @param options.transactions - The array of transactions within a batch request.
+ * @returns An object containing the transactions with their gas limits and the total gas limit.
+ */
+export async function simulateGasBatch({
+  chainId,
+  from,
+  transactions,
+}: {
+  chainId: Hex;
+  from: Hex;
+  transactions: TransactionBatchSingleRequest[];
+}): Promise<{ transactions: TransactionBatchSingleRequest[]; gasLimit: Hex }> {
+  const response = await simulateTransactions(chainId, {
+    transactions: transactions.map((transaction) => ({
+      ...transaction.params,
+      from,
+    })),
+  });
+
+  if (
+    !response?.transactions ||
+    response.transactions.length !== transactions.length
+  ) {
+    throw new Error('Simulation response does not match transaction count');
+  }
+
+  const { transactionsWithGasLimit, totalGasLimit } = transactions.reduce(
+    (acc, transaction, index) => {
+      const gasLimit = response.transactions[index]?.gasLimit;
+
+      if (!gasLimit) {
+        throw new Error(`No simulated gas returned`);
+      }
+
+      acc.totalGasLimit = acc.totalGasLimit.add(hexToBN(gasLimit));
+
+      acc.transactionsWithGasLimit.push({
+        ...transaction,
+        params: {
+          ...transaction.params,
+          gas: gasLimit,
+        },
+      });
+
+      return acc;
+    },
+    {
+      transactionsWithGasLimit: [] as TransactionBatchSingleRequest[],
+      totalGasLimit: new BN(0),
+    },
+  );
+
+  return {
+    transactions: transactionsWithGasLimit,
```

Comment:
What I meant was can we simplify this function and only return the `gasLimit` as that's all we need currently?